File: LeetCode/Easy/0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        int[] m = new int[10_000];
        m[1] = 1;
        m[2] = 2;

        for (int i = 3; i <= n; i++) {
            m[i] = m[i - 1] + m[i - 2];
        }

        return m[n];
    }
}

File: LeetCode/Easy/0070-climbing-stairs/0070-climbing-stairs.kt
class Solution {
    fun climbStairs(n: Int): Int {
        if (n <= 1) return 1
        var last = 1
        var beforeLast = 2

        for (i in 3..n) {
            val tmp = last + beforeLast
            last = beforeLast
            beforeLast = tmp
        }

        return beforeLast
    }
}
